### PR TITLE
Enhancement : enhance deploy button

### PR DESCRIPTION
**Pull request by RemyNtshaykolo**

### CODE REVIEW

File: javascript/release-button/action.yml [hunk 1]
```diff
@@ -4,6 +4,16 @@ author: 'Vibou'
...
     description: 'the deploy id'
+    required: true
+  deploy_env:
+    description: 'Which environment is targeted for the deployment'
+    required: false
+    default: 'prod'
+  deploy_model:
+    description: 'model or code'
+    required: false
+    default: 'code'
```
Comment (vibou) [inline javascript/release-button/action.yml:15]: Do not use model. use deploy_type with default value = ''
In your case you can set the value to 'model' afterward

File: javascript/release-button/lib/deploy_button.js [hunk 2]
```diff
@@ -15,12 +15,21 @@ Object.defineProperty(exports, "__esModule", { value: true });
 const core = __importStar(require("@actions/core"));
 const crypto_1 = __importDefault(require("crypto"));
 const querystring_1 = require("querystring");
-function generateButton(deployId) {
+var DeployEnv;
+(function (DeployEnv) {
+    DeployEnv["dev"] = "dev";
+    DeployEnv["preprod"] = "preprod";
+    DeployEnv["prod"] = "prod";
+})(DeployEnv || (DeployEnv = {}));
```
Comment (vibou) [inline javascript/release-button/lib/deploy_button.js:23]:
```suggestion
const DeployEnv = {};
(function (DeployEnv) {
    DeployEnv["dev"] = "dev";
    DeployEnv["preprod"] = "preprod";
    DeployEnv["prod"] = "prod";
})(DeployEnv);
```

File: javascript/release-button/src/deploy_button.ts [hunk 3]
```diff
@@ -4,19 +4,35 @@ import * as core from '@actions/core'
...
       'Environment variable PRIVATE_KEY, GITHUB_REPOSITORY and GITHUB_REF are required.',
     )
     process.exit(1)
-    return
   }
 
-  const [owner, repo] = GITHUB_REPOSITORY.split('/')
-  const [, , refName] = GITHUB_REF.split('/')
+  const [owner, repo] = GITHUB_REPOSITORY!.split('/')
+  const [, , refName] = GITHUB_REF!.split('/')
```
Comment (vibou) [inline javascript/release-button/src/deploy_button.ts:35]: if (!PRIVATE_KEY || !GITHUB_REPOSITORY || !GITHUB_REF) already check the undefined/null value so you should not have to unwrap the optional value here are you?

File: javascript/release-button/src/deploy_button.ts [hunk 4]
```diff
@@ -26,20 +42,43 @@ function generateButton(deployId: string) {
...
   process.stdout.write(`[![Deploy to prod](${img})](${url})`)
 }
 
 try {
-  const deployId = core.getInput('deploy_id', { required: true })
-  generateButton(deployId)
+  const deployId = core.getInput('deploy_id')
+  const deployModel = core.getInput('deploy_model')
+  const deployEnv = core.getInput('deploy_env')
+  generateButton(deployId, deployModel, deployEnv)
```
Comment (vibou) [inline javascript/release-button/src/deploy_button.ts:81]:
```suggestion
  const deployId = core.getInput('deploy_id')
  const deployType = core.getInput('deploy_type')
  const deployEnv = core.getInput('deploy_env')
  generateButton(deployId, deployType, deployEnv)
```

File: javascript/release-button/lib/deploy_button.js [hunk 5]
```diff
@@ -34,13 +43,27 @@ function generateButton(deployId) {
...
+            buttonStyle = { name: 'Dev', color: 'blue' };
+            break;
+        case DeployEnv.preprod:
+            buttonStyle = { name: 'Preprod', color: 'yellow' };
+            break;
+        case DeployEnv.prod:
+            buttonStyle = { name: 'Production', color: 'orange' };
+            break;
+    }
+    const img = `https://img.shields.io/badge/Deploy${deployType === 'model' ? '%20model' : ''}%20to-${buttonStyle.name}-${buttonStyle.color}.svg?style=for-the-badge`;
```
Comment (vibou) [inline javascript/release-button/lib/deploy_button.js:58]:
```suggestion
	const title = encodeURI(`Deploy ${deployType} to`.replace(/\s+/g,' '))
    const img = `https://img.shields.io/badge/${title}-${buttonStyle.name}-${buttonStyle.color}.svg?style=for-the-badge`;
```

**Review by vibou — CHANGES_REQUESTED** [review]
Some suggestions

File: javascript/release-button/src/deploy_button.ts (hunk 3 above)
Comment (RemyNtshaykolo) [reply javascript/release-button/src/deploy_button.ts:35]: I agree but I have an  "Object is possibly 'undefined'" error when compiling if I don't explicitly  specify the value as non undefined with "!".

File: javascript/release-button/lib/deploy_button.js (hunk 2 above)
Comment (RemyNtshaykolo) [reply javascript/release-button/lib/deploy_button.js:23]: should we modify the js file?

Comment (vibou) [reply javascript/release-button/lib/deploy_button.js:23]: no no i didn't see that it was generated version

File: javascript/release-button/src/deploy_button.ts (hunk 3 above)
Comment (vibou) [reply javascript/release-button/src/deploy_button.ts:35]: I was afraid of this answer :D

**Review by vibou — APPROVED** [review]

**Pull request merged**